src/tags.psm1

```diff
@@ -0,0 +1,1360 @@
+    if (-not $AssetPartitionId) {
+        $AssetPartitionId = -1
+    }
+    $local:RelPath = "AssetPartitions/$AssetPartitionId/Tags"
+
+    if ($PSBoundParameters.ContainsKey("Tag")) {
+        # tag specified, so get the ID based on name
+        $local:TagId = ( Resolve-SafeguardTagId -AccessToken $AccessToken -Appliance $Appliance -Insecure:$Insecure -AssetPartitionId $AssetPartitionId $Tag)
+        # get all assigned 
+        Invoke-SafeguardMethod -AccessToken $AccessToken -Appliance $Appliance -Insecure:$Insecure Core GET "$local:RelPath/$local:TagId/Occurrences" -Parameters $local:Parameters
+    }
+}
+
+
+<#
+.SYNOPSIS
+Get the tags from a specific asset via the REST-API.
+
+.DESCRIPTION
+Get the assigned tags for a specific asset.
+
+.PARAMETER Appliance
+IP address or hostname of a Safeguard appliance.
+
+.PARAMETER AccessToken
+A string containing the bearer token to be used with Safeguard Web API.
+
+.PARAMETER Insecure
+Ignore verification of Safeguard appliance SSL certificate.
+
+.PARAMETER AssetPartition
+An integer containing an ID or a string containing the name of the asset partition to get tags from an asset.
+If this value is empty, the Macrocosm partition will be used.
+
+.PARAMETER AssetPartitionId
+An integer containing the asset partition ID to get the asset's tags from.
+(If specified, this will override the AssetPartition parameter)
+
+.PARAMETER Asset
+Required parameter. An integer containing the ID of the asset or a string containing the name of the asset to get the tags for.
+
+.PARAMETER Fields
+An array of the tag property names to return.
+
+.INPUTS
+None.
+
+.OUTPUTS
+JSON response from Safeguard Web API.
+
+.EXAMPLE
+Get-SafeguardAssetTags "assetname" 
+
+.EXAMPLE
+Get-SafeguardAssetTags 14 
+
+.EXAMPLE
+Get-SafeguardAssetTags "assetname" -Fields Id,Name,Description
+
+#>
+function Get-SafeguardAssetTags {
+    [CmdletBinding()]
+    Param(
+        [Parameter(Mandatory=$false)]
+        [string]$Appliance,
+        [Parameter(Mandatory=$false)]
+        [object]$AccessToken,
+        [Parameter(Mandatory=$false)]
+        [switch]$Insecure,
+        [Parameter(Mandatory=$false)]
+        [object]$AssetPartition,
+        [Parameter(Mandatory=$false)]
+        [int]$AssetPartitionId = $null,
+        [Parameter(Mandatory=$true,Position=0)]
+        [object]$Asset,
+        [Parameter(Mandatory=$false)]
+        [string[]]$Fields
+    )
+
+    if (-not $PSBoundParameters.ContainsKey("ErrorAction")) { $ErrorActionPreference = "Stop" }
+    if (-not $PSBoundParameters.ContainsKey("Verbose")) { $VerbosePreference = $PSCmdlet.GetVariableValue("VerbosePreference") }
+
+    $local:Parameters = $null
+    if ($Fields) {
+        $local:Parameters = @{ fields = ($Fields -join ",")}
+    }
+    
+    # we need to use function Resolve-SafeguardAssetId from the asset.psm1 module to get the Asset ID so check if the module is load
+    if (-not (Get-Module assets)) {
```

Comment:
It might be easier to maintain in the long run if this code is separated out into an internal function.  I noticed it is repeated a few times.